components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/BPMNDataPublisher.java

```diff
@@ -205,6 +252,63 @@ public void configure() throws IOException, XMLStreamException, DataEndpointAuth
             } else {
                 log.warn("Required fields for data publisher are not configured. Receiver URLs, username and password are mandatory. Data publishing will not be enabled.");
             }
+    }
+
+    /**
+     * @param receiverURLSet
+     * @param username
+     * @param password
+     * @param authURLSet
+     * @param type
+     * @throws DataEndpointAuthenticationException
+     * @throws DataEndpointAgentConfigurationException
+     * @throws TransportException
+     * @throws DataEndpointException
+     * @throws DataEndpointConfigurationException
+     */
```

Comment:
Complete Java docs : Add information about usage of the function, parameters and exceptions